scripts/o3de/tests/test_repo.py

```diff
@@ -0,0 +1,205 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import argparse
+import copy
+import json
+import logging
+import pytest
+import pathlib
+import urllib.request
+from unittest.mock import patch, MagicMock, mock_open
+
+from o3de import manifest, register, repo
+
+TEST_O3DE_MANIFEST_JSON_PAYLOAD = '''
+{
+    "o3de_manifest_name": "testuser",
+    "origin": "C:/Users/testuser/.o3de",
+    "default_engines_folder": "C:/Users/testuser/.o3de/Engines",
+    "default_projects_folder": "C:/Users/testuser/.o3de/Projects",
+    "default_gems_folder": "C:/Users/testuser/.o3de/Gems",
+    "default_templates_folder": "C:/Users/testuser/.o3de/Templates",
```

Comment:
nit: the default folder  for projects, gems and templates should be in `O3DE` not `.o3de` per 
 https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/manifest.py#L40